### PR TITLE
refactor(*): consolidate

### PR DESCRIPTION
**Pull request by ckohen**
Some of this code was quite redundant, especially when adding new packages / apps

This refactor should make it 1 line changes to add new packages / apps (excluding readme docs).

Additionally, in order to support the refactor, the split to apps was made here. It should otherwise be identical to the functionality before the refactor.

### CODE REVIEW

**Review by Jiralite — APPROVED** [review]

File: src/api/_lib/utils/webhooks.ts [hunk 1]
```diff
@@ -37,23 +31,6 @@ export const DiscordWebhooks = useForum ? ForumWebhooks : Webhooks;
...
-export const PerPackageWebhooks: Record<PackageName, DiscordWebhooksTarget> = {
-	actions: 'monorepo',
-	brokers: 'brokers',
-	builders: 'builders',
-	collection: 'collection',
-	'discord.js': 'discord.js',
-	docgen: 'monorepo',
-	guide: 'guide',
-	proxy: 'proxy',
+export const OverideWebhooks: Partial<Record<AppName | PackageName, DiscordWebhooksTarget>> = {
```
Comment (kyranet) [inline src/api/_lib/utils/webhooks.ts:34]:
```suggestion
export const OverrideWebhooks: Partial<Record<AppName | PackageName, DiscordWebhooksTarget>> = {
```

And update the references everywhere else.

File: src/api/_lib/handlers/push.ts [hunk 2]
```diff
@@ -11,23 +12,34 @@ export function getPushRewriteTarget(event: PushEvent): DiscordWebhooksTarget {
 	// Workaround for pre-monorepo
 	if (event.ref === 'refs/heads/v13') return 'discord.js';
 
-	const packages = new Set<PackageName>();
+	const targets = new Set<AppName | PackageName>();
 
 	for (const commit of event.commits) {
+		for (const name of Object.values(AppName)) {
```
Comment (kyranet) [inline src/api/_lib/handlers/push.ts:18]: I noticed we do `Object.values(AppName)` and `Object.values(PackageName)` very frequently, can't we cache them under `AppNameValues` and `PackageNameValues`?

File: src/api/_lib/handlers/push.ts [hunk 3]
```diff
@@ -11,23 +12,34 @@ export function getPushRewriteTarget(event: PushEvent): DiscordWebhooksTarget {
...
+			if (commit.added.some((file) => file.startsWith(`apps/${name}/`))) {
+				targets.add(name);
+			}
+			if (commit.modified.some((file) => file.startsWith(`apps/${name}/`))) {
+				targets.add(name);
+			}
+			if (commit.removed.some((file) => file.startsWith(`apps/${name}/`))) {
+				targets.add(name);
+			}
```
Comment (kyranet) [inline src/api/_lib/handlers/push.ts:27]:
```suggestion
			const prefix = `apps/${name}/`;
			if (commit.added.some((file) => file.startsWith(prefix))) {
				targets.add(name);
			}
			if (commit.modified.some((file) => file.startsWith(prefix))) {
				targets.add(name);
			}
			if (commit.removed.some((file) => file.startsWith(prefix))) {
				targets.add(name);
			}
```

Comment (kyranet) [reply src/api/_lib/handlers/push.ts:27]: Same thing would apply to the loop below.

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/api/_lib/handlers/push.ts [hunk 4]
```diff
@@ -1,6 +1,19 @@
...
+
+/**
+ * Checks if a paths files were modified
+ * @param prefix The paths whos files to check
+ * @param commit The commit details
+ */
+function checkModified(prefix: string, commit: Commit) {
+	return (
+		commit.added.some((file) => file.startsWith(prefix)) ||
+		commit.modified.some((file) => file.startsWith(prefix) || commit.removed.some((file) => file.startsWith(prefix)))
```
Comment (kyranet) [inline src/api/_lib/handlers/push.ts:14]: You have one `)` in the wrong place.

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/api/_lib/handlers/push.ts (hunk 4 above)
Comment (ckohen) [reply src/api/_lib/handlers/push.ts:14]: smh lint did me dirty

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**